vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/MenuContentBuilder.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.contextmenu;
+
+import java.io.Serializable;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.dom.Element;
+
+/**
+ * A class which is utilized internally by components such as context menu and
+ * menu bar. It transform the components added to the menus and their sub menus
+ * to the hierarchical items array of the web component.
+ *
+ * @param <I>
+ *            the menu item type
+ */
+public class MenuContentBuilder<I extends MenuItemBase<?, I, ?>>
+        implements Serializable {
+
+    private Component menu;
+
+    private boolean updateScheduled = false;
+    private final Element container;
+
+    public MenuContentBuilder(Component menu) {
+        this.menu = menu;
+        container = new Element("div");
+        getElement().appendVirtualChild(container);
+
+        getElement().getNode().runWhenAttached(ui -> ui.getPage()
+                .addJavaScript("frontend://contextMenuConnector.js"));
+    }
+
+    /**
+     * Rebuilds the client-side items array with the current components in the
+     * menu and its sub menus.
+     */
+    public void build() {
```

Comment:
Even though not sure about the naming of `build` as it's taking care of `reseting the Content`
Not a blocker.

Reply:
renamed